core/modules/parsers/videoparser.js

```diff
@@ -6,29 +6,161 @@ module-type: parser
 The video parser parses a video tiddler into an embeddable HTML element
 
 \*/
-(function(){
+(function () {
 
 /*jslint node: true, browser: true */
 /*global $tw: false */
 "use strict";
 
-var VideoParser = function(type,text,options) {
+var VideoParser = function (type, text, options) {
 	var element = {
+		type: "element",
+		tag: "video",
+		attributes: {
+			controls: { type: "string", value: "controls" },
+			style: { type: "string", value: "width: 100%; object-fit: contain" },
+			preload: { type: "string", value: "auto" },
+			class: { type: "string", value: "tw-video-element" }
+		}
+	};
+
+	if (options._canonical_uri) {
+		element.children = [{
 			type: "element",
-			tag: "video",
+			tag: "source",
 			attributes: {
-				controls: {type: "string", value: "controls"},
-				style: {type: "string", value: "width: 100%; object-fit: contain"}
+				src: { type: "string", value: options._canonical_uri },
+				type: { type: "string", value: type }
 			}
-		},
-		src;
-	if(options._canonical_uri) {
-		element.attributes.src = {type: "string", value: options._canonical_uri};
-	} else if(text) {
-		element.attributes.src = {type: "string", value: "data:" + type + ";base64," + text};
+		}];
+	} else if (text) {
+		element.attributes.src = { type: "string", value: "data:" + type + ";base64," + text };
+	}
+
+	if ($tw.browser) {
+		const processedVideos = new WeakMap();
```

Comment:
@Jermolene -- Do we allow `WeakMap` ES6 (ES2015) and `const`, `async` and `arrow => functions`?